torchvision/prototype/datasets/_builtin/stanford_cars.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any, Dict, List
+
+from torchdata.datapipes.iter import Filter, IterDataPipe, Mapper, Zipper
+from torchvision.prototype.datasets.utils import Dataset, DatasetConfig, DatasetInfo, HttpResource, OnlineResource
+from torchvision.prototype.datasets.utils._internal import hint_sharding, hint_shuffling, path_comparator, read_mat
+from torchvision.prototype.features import BoundingBox, EncodedImage
+
+
+class StanfordCars(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            name="stanford_cars",
+            homepage="https://ai.stanford.edu/~jkrause/cars/car_dataset.html",
+            dependencies=("scipy",),
+            valid_options=dict(
+                split=("test", "train"),
+            ),
+        )
+
+    _URL_ROOT = "https://ai.stanford.edu/~jkrause/"
+    _URLS = {
+        "train": f"{_URL_ROOT}car196/cars_train.tgz",
+        "test": f"{_URL_ROOT}car196/cars_test.tgz",
+        "test_ground_truth": f"{_URL_ROOT}car196/cars_test_annos_withlabels.mat",
+        "devkit": f"{_URL_ROOT}cars/car_devkit.tgz",
+    }
+
+    _CHECKSUM = {
+        "train": "512b227b30e2f0a8aab9e09485786ab4479582073a144998da74d64b801fd288",
+        "test": "bffea656d6f425cba3c91c6d83336e4c5f86c6cffd8975b0f375d3a10da8e243",
+        "test_ground_truth": "790f75be8ea34eeded134cc559332baf23e30e91367e9ddca97d26ed9b895f05",
+        "devkit": "512b227b30e2f0a8aab9e09485786ab4479582073a144998da74d64b801fd288",
+    }
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        resources = [HttpResource(self._URLS[config.split], sha256=self._CHECKSUM[config.split])]
+        if config.split == "test":
+            resources.append(HttpResource(self._URLS["test_ground_truth"], sha256=self._CHECKSUM["test_ground_truth"]))
+        else:
+            resources.append(HttpResource(url=self._URLS["devkit"], sha256=self._CHECKSUM["devkit"]))
+        return resources
+
+    def _make_datapipe(
+        self,
+        resource_dps: List[IterDataPipe],
+        *,
+        config: DatasetConfig,
+    ) -> IterDataPipe[Dict[str, Any]]:
+
+        images_dp, targets_dp = resource_dps
+        print(config.split)
+        if config.split == "train":
+            targets_dp = Filter(targets_dp, path_comparator("name", "cars_train_annos.mat"))
+        dp = Zipper(images_dp, targets_dp)
```

Comment:
This is dangerous. Did you make sure that both datapipes are perfectly aligned? Otherwise we should use an `IterKeyZipper`.

Reply:
I wasn't aware of this! I will look into it. Thanks!

Reply:
I think that zipping these two pipelines is not a good idea; Because `targets_dp` contains only one file and images_dp contains around 8000 images;  So with the current implementation, we can access only one image.. 

I was thinking about decoding the `targets_dp` and saving the target_dp data into a variable (something like `self.labels` ) and then accessing it whenever needed...

Could you give me some feedback regarding this? Like should i go with this? Or is there any other way to tackle this..

Reply:
They are aligned :)